### PR TITLE
UPD of developer's guide and docs for materials

### PR DESCRIPTION
**Pull request by KulaginVladimir**
## Proposed changes

Updates docs according to #884, #852, and #841 

@RemDelaporteMathurin, let me know how the docs could be improved

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [ ] Black formatted
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by RemDelaporteMathurin — COMMENTED** [review]
Thanks a lot for this Vladimir! Grinding towards 1.4!!

File: docs/source/devguide/index.rst [hunk 1]
```diff
@@ -217,4 +246,9 @@ When contributing to the documentation, make sure to:
 #. Write clear and concise documentation
 #. Use the right syntax
 #. Update the documentation when new features are added
+#. Test the documentation using:
+
+.. code-block:: bash
 
+   cd docs/source
+   make doctest
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply docs/source/devguide/index.rst:254]: @RemDelaporteMathurin 
`sphinx-build -b doctest source build` seems to work

File: docs/source/devguide/index.rst [hunk 2]
```diff
@@ -66,6 +66,11 @@ For complete information on contributions with GitHub see this guide on `GitHub
    You may want to build the documentation to see if your changes are correctly reflected or if you are updating the docs.
    See :ref:`Documentation guide` for more information.
 
+#. Optional: Format your code using `Black <https://github.com/psf/black>`_.
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply docs/source/devguide/index.rst]: Removed "optional"

File: docs/source/devguide/index.rst [hunk 3]
```diff
@@ -158,6 +163,30 @@ Implementing a new feature
 #. Open a PR
 
 
+----------------
+Code formatting
+----------------
+
+Before merging your PR, the modified scripts should be formatted to maintain the consistency of the coding style. FESTIM is formatted using 
+`Black <https://github.com/psf/black>`_ - the uncompromising Python code formatter. To install Black, run the following command:
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply docs/source/devguide/index.rst]: Added, the guide is provided on the page of the extension (also in the VS code extensions marketplace)

File: docs/source/userguide/materials.rst [hunk 4]
```diff
@@ -14,34 +14,77 @@ To define a material, use the :class:`festim.Material` class:
...
+
+    my_mat = Material(
+        id=1,
+        D_0=2e-7,
+        E_D=0.2,
+        thermal_cond=lambda T: 3 * T + 2,
+        heat_capacity=lambda T: 4 * T + 8,
+        rho=lambda T: 7 * T + 5,
+        Q=lambda T: -0.5 * T**2,
+    )
```
[the remark this answers is not in the document]
Comment (KulaginVladimir) [reply docs/source/userguide/materials.rst]: I modified the example

File: docs/source/devguide/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply docs/source/devguide/index.rst:257]: can we add it as an alternative?

[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply docs/source/devguide/index.rst:257]: aready done mb

**Review by RemDelaporteMathurin — APPROVED** [review]

**Pull request merged**